src/game_api/script/lua_vm.cpp

```diff
@@ -2260,6 +2260,16 @@ end
     /// Initializes some seedeed run related values and loads the character select screen, as if starting a new seeded run after entering the seed.
     lua["play_seeded"] = init_seeded;
 
+    lua.script(R"##(
+        function deepcopy_object(obj)
+            if type(obj) ~= 'table' then return obj end
+            local res = {}
+            for k, v in pairs(obj) do res[deepcopy_object(k)] = deepcopy_object(v) end
+            res = setmetatable(res, getmetatable(obj))
+            return res
+        end
+    )##");
+
```

Comment:
Are you using this anywhere? or it's just new addition?
does it even work? i know some weird objects like std::vector, or stuff that i did for the global `players` do not list anything with `pairs`

Reply:
I don't think it would work well with any object other than lua primitives and tables

Reply:
The part that I don't like is that every time you create a new table, you make new allocations, this means when playing online the object in `state.user_data` and the tables inside it will make many new unnecessary allocations per frame on this function, but I can't think any way of solving this rather than creating a new lua type.
I guess another way would be to not initialize new tables when it already exists, and just copying the primitives in it.
Though, I guess it probably doesn't affect performance much anyway and it's just me wanting to do this very optimized

Reply:
Oh I didn't read the first part, it is used [here](https://github.com/spelunky-fyi/overlunky/pull/369/files#diff-1162760f250e1dbe80e6b009f3c1201e8b7c46abef51b67df4535f1c169a444fR1891)

Reply:
I guess I should change [this](https://github.com/spelunky-fyi/overlunky/pull/369/files#diff-d3d0e735e82684d74e7cc32e6a04d7b32540afe19fd45f1c2dd5643752b1fa19R3087) description to say it won't work well with C++ objects then